### DISCUSSION TITLE
[Feature] Allow big file chunking when uploading

### DISCUSSION THREAD

**Discussion by daddyparodz** [immich-app/immich#2357] [Feature Request]
### The feature

We often have files that are way too big (over 100mb) and we can't upload them using Cloudflare tunnel or DNS (because they limit their maximum POST upload size), at least it didn't work for me.
I've already tried using local ip uploading and it works just fine with any filesize.
Could you add an option to break up big files into smaller "chunks" during the upload or at least help me solve this issue?
I really like this app, keep up the good work!

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by samip5** [comment]
That's free Cloudflare for ya as it has limits which non paying user cannot change.

And I think the mobile client is able to upload files larger than 100M currently.

**Comment by oblivioncth** [comment]
This is a duplicate of https://github.com/immich-app/immich/discussions/1674

As samip5 mentioned, a workaround was merged into the client that should avoid the cloudflare limit, although it's not actually chunking the upload so if cloudflare changes some of their internal handling of the request then the workaround *may* stop working.

Ultimately true chunking via TUS should be implemented in the long run.
